pypeman/tests/test_ctx_nodes.py

```diff
@@ -0,0 +1,91 @@
+"""
+tests for pypeman.contrib.ctx
+"""
+
+import asyncio
+
+import pytest
+
+from pypeman.contrib.ctx import CombineCtx
+from pypeman.nodes import NodeException
+from pypeman.tests.pytest_helpers import clear_graph  # noqa: F401
+
+from pypeman.tests.common import generate_msg
+# TODO: might refactor to another file?
+from pypeman.tests.test_nodes import FakeChannel
+
+
+def mk_msgs_w_ctx(*ctx_ids):
+    """
+    helper to create a msg with a few contexts
+    """
+    msg = generate_msg(
+        message_content="test",
+        message_meta=dict(entry1="meta1"),
+        )
+    ctx_msgs = []
+    for ctx_id in ctx_ids:
+        meta = dict(entry1="meta_%s" % ctx_id)
+        ctx_msg = generate_msg(
+            message_content={"val_%s" % ctx_id: "data_%s" % ctx_id},
+            message_meta=meta,
+            )
+        msg.add_context(ctx_id, ctx_msg)
+        ctx_msgs.append(ctx_msg)
+
+    return msg, ctx_msgs
+
+
+#    def __init__(self, ctx_names, meta_from=None, flatten=False,  *args, **kwargs):
```

Comment:
Do you want to keep this ?

Reply:
no. will remove

Reply:
779381ec991815ffbc9519b4c0cfc650e8950448